src/avk.cpp

```diff
@@ -7129,11 +7129,76 @@ namespace avk
 		return create_semaphore(device(), dispatch_loader_core(), std::move(aAlterConfigBeforeCreation));
 	}
 
+	semaphore root::create_timeline_semaphore(uint64_t aPayload, std::function<void(semaphore_t&)> aAlterConfigBeforeCreation)
+	{
+		auto typeInfo = std::make_unique<vk::SemaphoreTypeCreateInfo>();
+		typeInfo->semaphoreType = vk::SemaphoreType::eTimeline;
+		typeInfo->initialValue = aPayload;
+
+		return create_semaphore(device(), dispatch_loader_core(), [otherAlterations = move(aAlterConfigBeforeCreation), &typeInfo](semaphore_t& aSem) {
+			aSem.create_info().pNext = typeInfo.get();
+
+			if (otherAlterations) {
+				otherAlterations(aSem);
+			}
+		});
+	}
+
 	semaphore_t& semaphore_t::handle_lifetime_of(any_owning_resource_t aResource)
 	{
 		mLifetimeHandledResources.push_back(std::move(aResource));
 		return *this;
 	}
+
+	const uint64_t semaphore_t::query_current_value() const {
+		uint64_t value;
+		auto result = mSemaphore.getOwner().getSemaphoreCounterValue(mSemaphore.get(), &value);
+		assert(static_cast<VkResult>(result) >= 0);
+		return value;
+	}
+
+	void semaphore_t::signal(uint64_t aNewValue) const {
+
+		vk::SemaphoreSignalInfo info{};
+		info.sType = vk::StructureType::eSemaphoreSignalInfo;
+		info.pNext = NULL;
+		info.semaphore = mSemaphore.get();
+		info.value = aNewValue;
+
+		mSemaphore.getOwner().signalSemaphore(info);
+	}
+
+	vk::Result semaphore_t::wait_until_signaled(uint64_t aRequiredValue, std::optional<uint64_t> aTimeout) const {
+		std::vector<std::reference_wrapper<const semaphore_t>> semaphores{ *this };
+		return wait_until_signaled(semaphores, std::span<uint64_t>(&aRequiredValue, 1), true, aTimeout);

```

Comment:
Construction of `std::span` looks veeeery odd here. It actually defies the purpose of a `std::span`. I.e, I would say, the point is to never explicitly create a `std::span`, but use it to refer to just elements in contiguous memory (wherever they might be stored, e.g., `std::array` or `std::vector`, ...)